moveit_ros/benchmarks/src/BenchmarkExecutor.cpp

```diff
@@ -897,6 +922,95 @@ void BenchmarkExecutor::collectMetrics(PlannerRunData& metrics,
   }
 }
 
+void BenchmarkExecutor::computeResultPathSimilarity(PlannerBenchmarkData& planner_data,
+                                                    const std::vector<planning_interface::MotionPlanDetailedResponse>& mp_res,
+                                                    const std::vector<bool>& solved)
+{
+  ROS_INFO("Computing result path similarity");
+  const size_t result_count = planner_data.size();
+  size_t unsolved = std::count_if(solved.begin(), solved.end(), [](bool s){return !s;});
+  std::vector<double> average_distances(mp_res.size());
+  for (size_t i = 0; i < result_count; ++i)
+  {
+    if (!solved[i])
+    {
+      average_distances[i] = std::numeric_limits<double>::max();
+      continue;
+    }
+    for (size_t j = i+1; j < result_count; ++j)
+    {
+      if (!solved[j])
+        continue;
+
+      // Abort if there are no trajectories
+      if (mp_res[i].trajectory_.empty() || mp_res[j].trajectory_.empty())
+        continue;
+
+      // Access trajectories
+      const robot_trajectory::RobotTrajectory& traj_first = *mp_res[i].trajectory_.back();
+      const robot_trajectory::RobotTrajectory& traj_second = *mp_res[j].trajectory_.back();
+
+      // Abort if trajectories are empty
+      if (traj_first.empty() || traj_second.empty())
+        continue;
+
+      // Waypoint counter
+      size_t pos_first = 0;
+      size_t pos_second = 0;
+      const size_t max_pos_first = traj_first.getWayPointCount() - 1;
+      const size_t max_pos_second = traj_second.getWayPointCount() - 1;
+
+      // compute average distance
+      double average_distance = 0;
+      size_t steps = 0;
+      double current_distance = traj_first.getWayPoint(pos_first).distance(traj_second.getWayPoint(pos_second));
+      while(true)
+      {
```

Comment:
the contents of this while should probably be in a helper function, due to excess indentation

Reply:
I'm not sure about this since it would complicate handling the variables and we don't reuse any of the code.

Reply:
code re-use isn't a requirement of helper functions, rather readability and modularization of utility. but if you really don't like the idea, im fine with that.

Reply:
Instead of factoring out the while loop, I suggest introducing a function 
`computeTrajectoryDistance(traj_first, traj_second)`

Reply:
Please add a comment describing your distance measuring approach. Something like:
> This function aligns both trajectories in a greedy fashion and computes the mean waypoint distance averaged over all aligned waypoints. Using a greedy approach is more efficient than dynamic time warping, and seems to be sufficient for similar trajectories.